sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/metadata/TableMetaDataLoader.java

```diff
@@ -205,9 +207,9 @@ private boolean isTableExist(final Connection connection, final String catalog,
         return Optional.of(new ColumnMetaData(columnName, columnType, isPrimaryKey, isNotNull, isAutoIncrement));
     }
     
-    private Collection<String> getLogicIndexes(final Connection connection, final String catalog, final String actualTableName) throws SQLException {
+    private Collection<String> getLogicIndexes(final Connection connection, final String catalog, final String actualTableName, final String dataSourceName) throws SQLException {
         Collection<String> result = new HashSet<>();
-        try (ResultSet resultSet = connection.getMetaData().getIndexInfo(catalog, catalog, actualTableName, false, false)) {
+        try (ResultSet resultSet = connection.getMetaData().getIndexInfo(catalog, dataSourceName, actualTableName, false, false)) {
```

Comment:
Hi, why do you use datasource instead of catalog?
dataSourceName, which is configured by user, may be different with catalog.

Reply:
emmm, That's true. the schemaName is allow Null, I didn't think about it  may be different with catalog. And this problem is fix. I will close this pr